src/syft/lib/torch/allowlist.py

```diff
@@ -482,7 +482,7 @@
 allowlist["torch.Tensor.tan"] = "torch.Tensor"
 # allowlist['torch.Tensor.tan_'] = #
 allowlist["torch.Tensor.tanh"] = "torch.Tensor"
-# allowlist['torch.Tensor.tanh_'] = #
+allowlist["torch.Tensor.tanh_"] = "torch.Tensor"
```

Comment:
tanh_ returns None not torch.Tensor

Reply:
I spent some time hunting through the Pytorch code which is difficult but most of the functionality terminates in ATen.

https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/UnaryOps.cpp
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/BinaryOps.cpp

All the _ functions I can find seem to follow the same pattern:

```c
// UnaryOps.cpp
Tensor tanh(const Tensor& self) { return unary_op_impl(self, at::tanh_out); }
Tensor& tanh_(Tensor& self) { return unary_op_impl_(self, at::tanh_out); }
```

```c
// BinaryOps.cpp
Tensor div(const Tensor& self, const Tensor& other) {
  if (isIntegralType(self.scalar_type(), /*includeBool=*/ true)
      && isIntegralType(other.scalar_type(), /*includeBool=*/ true)) {
    TORCH_CHECK(false,
      "Integer division of tensors using div or / is no longer supported, ",
      "and in a future release div will perform true division as in Python 3. ",
      "Use true_divide or floor_divide (// in Python) instead.");
  }

  Tensor result;
  auto iter = TensorIterator::binary_op(result, self, other);
  div_stub(iter.device_type(), iter);
  return iter.output();
}

Tensor& div_(Tensor& self, const Tensor& other) {
  return native::div_out(self, self, other);
}
```

They are the same as their non _ versions except that they take a non const reference and return a reference. So they all do return something and it seems that when this translates to python the result is it returns the "self" which was used for the operation.

<img width="461" alt="Screen Shot 2020-08-26 at 9 28 28 am" src="https://user-images.githubusercontent.com/2882739/91249766-dacbf180-e79a-11ea-8fb0-31ba02c4a4cf.png">

Do we want to do anything different to this?